src/ebay-listbox-button/listbox-button.tsx

```diff
@@ -172,7 +184,7 @@ const ListboxButton: FC<EbayListboxButtonProps> = ({
                 setExpanded(false)
                 setTimeout(() => setSelectedOption(childrenArray[selectedIndex]))
                 setTimeout(() => buttonRef.current.focus(), 0)
-                onSelect(e, getSelectedValueByIndex(selectedIndex), selectedIndex)
+                onChange(e as any, { index: selectedIndex, selected: [String(selectedIndex)], wasClicked })
```

Comment:
I think is `getSelectedValueByIndex(selectedIndex)` instead of `String(selectedIndex)`?